### PR TITLE
don't send packets after receiving a CONNECTION_CLOSE

### LINKED ISSUE #2374: don't acknowledge the CONNECTION_CLOSE
Occasionally, I'm seeing logs like these:
```
2020/02/24 11:38:21.406030 client -> Sending coalesced packet (1 parts, 1200 bytes) for connection 0x9d0ceeb448b3d7cc
2020/02/24 11:38:21.406033 client 	Long Header{Type: Initial, DestConnectionID: 0x931a1c7a, SrcConnectionID: (empty), Token: 0x51d75aff6d91599d51d416ecb968902592ca8b90d2b9f50898a8b5dec91d7b848107f1b03afacce9d74d9d57a5ffd564a3de057ffa9b95283e2b88833b4a6445e5461ae043b7878c73f25c14fa62c341, PacketNumber: 0x1, PacketNumberLen: 4, Length: 1105, Version: TLS dev version (WIP)}
2020/02/24 11:38:21.406035 client 	-> &wire.CryptoFrame{Offset: 0x0, Data length: 0x15e, Offset + Data length: 0x15e}
2020/02/24 11:38:21.406376 client <- Reading packet 0x0 (37 bytes) for connection (empty), Initial
2020/02/24 11:38:21.406379 client 	Long Header{Type: Initial, DestConnectionID: (empty), SrcConnectionID: 0x931a1c7a, Token: (empty), PacketNumber: 0x0, PacketNumberLen: 4, Length: 24, Version: TLS dev version (WIP)}
2020/02/24 11:38:21.406385 client 	<- &wire.ConnectionCloseFrame{IsApplicationError:false, ErrorCode:0xb, FrameType:0x0, ReasonPhrase:""}
2020/02/24 11:38:21.406389 client Peer closed session with error: INVALID_TOKEN
2020/02/24 11:38:21.406393 client 	Queueing ACK because the first packet should be acknowledged.
2020/02/24 11:38:21.406401 client -> Sending coalesced packet (1 parts, 1200 bytes) for connection 0x9d0ceeb448b3d7cc
2020/02/24 11:38:21.406405 client 	Long Header{Type: Initial, DestConnectionID: 0x931a1c7a, SrcConnectionID: (empty), Token: 0x51d75aff6d91599d51d416ecb968902592ca8b90d2b9f50898a8b5dec91d7b848107f1b03afacce9d74d9d57a5ffd564a3de057ffa9b95283e2b88833b4a6445e5461ae043b7878c73f25c14fa62c341, PacketNumber: 0x2, PacketNumberLen: 4, Length: 1105, Version: TLS dev version (WIP)}
2020/02/24 11:38:21.406410 client 	-> &wire.AckFrame{LargestAcked: 0x0, LowestAcked: 0x0, DelayTime: 0s}
2020/02/24 11:38:21.406416 client Connection 0x9d0ceeb448b3d7cc closed.
```

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #2374.

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -539,6 +539,12 @@ runLoop:
 			s.handleHandshakeComplete()
 		}
 
+		select {
```
Comment (lucas-clemente) [inline session.go]: Comment?

Comment (marten-seemann) [reply session.go]: Added a comment. Also moved this block up to the `case` that's handling received packets.

**Comment by marten-seemann** [comment]
This PR is wrong. It prevents us from sending any packet after the session was closed. This is only correct in the case where we receive the CONNECTION_CLOSE frame (i.e. where the session is closed remotely). It is not correct when we close the session locally.

**Comment by marten-seemann** [comment]
> This PR is wrong. It prevents us from sending any packet after the session was closed. This is only correct in the case where we receive the CONNECTION_CLOSE frame (i.e. where the session is closed remotely). It is not correct when we close the session locally.

That's not correct. The CONNECTION_CLOSE is sent in `handleCloseError`: https://github.com/lucas-clemente/quic-go/blob/49c6dfe2da82958589146a9b25925ef4920e1c8b/session.go#L1125-L1154

`handleCloseError` is called after the `runLoop` exits.

The reason CI is failing is that the CONNECTION_CLOSE is (occasionally) sent in a Handshake packet. The CI failure would thus be fixed by #2369. It's something worth looking into though, since it's weird that qtls gives us the keys even though the error occurred.

**Comment by marten-seemann** [comment]
> The reason CI is failing is that the CONNECTION_CLOSE is (occasionally) sent in a Handshake packet. The CI failure would thus be fixed by #2369. It's something worth looking into though, since it's weird that qtls gives us the keys even though the error occurred.

See #2386.

**Pull request merged**